src/foam/nanos/jetty/HttpServer.js

```diff
@@ -214,6 +241,50 @@ foam.CLASS({
           }
         });
       `
+    },
+    {
+      name: 'configHttps',
+      documentation: 'https://docs.google.com/document/d/1hXVdHjL8eASG2AG2F7lPwpO1VmcW2PHnAW7LuDC5xgA/edit?usp=sharing',
+      args: [
+        {
+          name: 'server',
+          javaType: 'final org.eclipse.jetty.server.Server'
+        }
+      ],
+      javaCode: `
+      if ( this.getEnableHttps() ) {
+  
+        try {
+          // 1. load the keystore to verify the keystore path and password.
+          KeyStore keyStore = KeyStore.getInstance("JKS");
+          keyStore.load(new FileInputStream(this.getKeystorePath()), this.getKeystorePassword().toCharArray());
+  
+          // 2. enable https
+          HttpConfiguration https = new HttpConfiguration();
+          https.addCustomizer(new SecureRequestCustomizer());
+          SslContextFactory sslContextFactory = new SslContextFactory();
+          sslContextFactory.setKeyStorePath(this.getKeystorePath());
+          sslContextFactory.setKeyStorePassword(this.getKeystorePassword());
+  
+          ServerConnector sslConnector = new ServerConnector(server,
+            new SslConnectionFactory(sslContextFactory, "http/1.1"),
+            new HttpConnectionFactory(https));
+          sslConnector.setPort(this.getHttpsPort());
+  
+          server.addConnector(sslConnector);
+  
+        } catch ( java.io.FileNotFoundException e ) {
+          System.err.println("No KeyStore file found at path: " + this.getKeystorePath() + e);
```

Comment:
Do we not have a context and logging?  Please output to both if we have access to the context here.  We'll be deploying this remotely.